### PR TITLE
add "some warnings detected" to ParseError printing when all problems are warnings

### PR DESCRIPTION
**Pull request by ericphanson**
I totally missed that the `(..)` issue from #350 was a warning rather than an error; I think probably bc the bold red part says `ERROR: ParseError:` and then the very light part says "Warning". Here I propose add to text to make it clearer by emphasizing warning vs error on the first line.

### CODE REVIEW

**Comment by c42f** [comment]
Do we need this for errors? It seems kind of redundant in that case.

Perhaps we just add the extra text when the error is due only to parser warnings?

File: src/parser_api.jl [hunk 1]
```diff
@@ -16,13 +16,16 @@ function ParseError(stream::ParseStream; incomplete_tag=:none, kws...)
...
 function Base.showerror(io::IO, err::ParseError)
-    println(io, "ParseError:")
     # Only show the first parse error for now - later errors are often
     # misleading due to the way recovery works
     i = findfirst(is_error, err.diagnostics)
     if isnothing(i)
         i = lastindex(err.diagnostics)
+        level_info = " some warnings detected:"
+    else
+        level_info = " some errors detected:"
```
Comment (ericphanson) [inline src/parser_api.jl:26]:
```suggestion
        level_info = ""
```

File: test/hooks.jl [hunk 2]
```diff
@@ -117,7 +117,7 @@ end
         if JuliaSyntax._has_v1_10_hooks
             @test err.args[1] isa Meta.ParseError
             exc = err.args[1]
-            @test exc.msg == "ParseError:\n# Error @ none:1:2\n\"\n#└ ── unterminated string literal"
+            @test exc.msg == "ParseError: some errors detected:\n# Error @ none:1:2\n\"\n#└ ── unterminated string literal"
```
Comment (ericphanson) [inline test/hooks.jl:120]:
```suggestion
            @test exc.msg == "ParseError\n# Error @ none:1:2\n\"\n#└ ── unterminated string literal"
```

File: test/parser_api.jl [hunk 3]
```diff
@@ -122,14 +122,14 @@ end
     catch exc
         @test exc isa JuliaSyntax.ParseError
         @test sprint(showerror, exc) == """
-            ParseError:
+            ParseError: some errors detected:
```
Comment (ericphanson) [inline test/parser_api.jl:125]:
```suggestion
            ParseError:
```

File: test/parser_api.jl [hunk 4]
```diff
@@ -122,14 +122,14 @@ end
...
             # Error @ somefile.jl:1:3
             a -- b -- c
             # └┘ ── invalid operator"""
         @test occursin("Stacktrace:\n", sprint(showerror, exc, catch_backtrace()))
         file_url = JuliaSyntax._file_url("somefile.jl")
         @test sprint(showerror, exc, context=:color=>true) == """
-            ParseError:
+            ParseError: some errors detected:
```
Comment (ericphanson) [inline test/parser_api.jl:132]:
```suggestion
            ParseError:
```

File: test/parser_api.jl [hunk 5]
```diff
@@ -145,7 +145,7 @@ end
     catch exc
         @test exc isa JuliaSyntax.ParseError
         @test sprint(showerror, exc) == """
-            ParseError:
+            ParseError: some errors detected:
```
Comment (ericphanson) [inline test/parser_api.jl:148]:
```suggestion
            ParseError:
```

File: test/hooks.jl [hunk 6]
```diff
@@ -117,7 +117,7 @@ end
         if JuliaSyntax._has_v1_10_hooks
             @test err.args[1] isa Meta.ParseError
             exc = err.args[1]
-            @test exc.msg == "ParseError:\n# Error @ none:1:2\n\"\n#└ ── unterminated string literal"
+            @test exc.msg == "ParseError\n# Error @ none:1:2\n\"\n#└ ── unterminated string literal"
```
Comment (ericphanson) [inline test/hooks.jl:120]:
```suggestion
            @test exc.msg == "ParseError:\n# Error @ none:1:2\n\"\n#└ ── unterminated string literal"
```

**Review by c42f — APPROVED** [review]

**Comment by c42f** [comment]
@ericphanson I invited you as a collaborator on this repo (if you like) feel free to merge this when you're ready :)

**Pull request merged**